### PR TITLE
Flavor Template Feature

### PR DESCRIPTION
**Pull request by aishwaryabk**
Fixes: #127
Signed-off-by: Cloud User <Aishwarya.Kamat@ibm.com>

### CODE REVIEW

File: openshift-install-powervs [hunk 1]
```diff
@@ -60,9 +61,9 @@ EOF
 
 RELEASE_VER=${RELEASE_VER:-"4.6"}
 ARTIFACTS_REPO=${ARTIFACTS_REPO:-"https://github.com/ocp-power-automation/ocp4-upi-powervs"}
-ARTIFACTS_VERSION=${ARTIFACTS_VERSION:-"release-$RELEASE_VER"}
+#ARTIFACTS_VERSION=${ARTIFACTS_VERSION:-"release-$RELEASE_VER"}
```
Comment (yussufsh) [inline openshift-install-powervs:64]: Why this change? This is not needed. Master is used only for dev preview releases.

File: openshift-install-powervs [hunk 2]
```diff
@@ -1422,6 +1477,12 @@ function main {
       warn "Enabling forceful destruction option for terraform destroy command"
       FORCE_DESTROY=1
       ;;
+    "-flavor")
+      shift
+      flavortype="$1"
+      vars+=" -var-file ../$varfile -var-file ../automation/compute-vars/$flavortype.tfvars"
```
Comment (yussufsh) [inline openshift-install-powervs:1483]: What if I give `-flavor` option before `-varfile` ? I feel `varfile` will be "".

File: openshift-install-powervs [hunk 3]
```diff
@@ -1328,6 +1363,26 @@ function process_varfile {
...
+#-------------------------------------------------------------------------
+function process_flavor {
+       flavor_file="$flavortype.tfvars"
+        [[ -z "$flavortype" ]] &&  error "The -flavor '$flavourtype' Please specify an option with flavor."
+
+       if [[ $flavortype == "small" || $flavortype == "medium" || $flavortype == "large" ]]; then
+          {
+              echo "IF_FLAVOR=2"
+              echo "FLAVOR_TYPE=$flavortype"
+          }  > logs/flavor.log
```
Comment (yussufsh) [inline openshift-install-powervs:1377]: No need to add it to logs.. logs are mostly TF console. You can use `log` function to display flavor on console.

File: openshift-install-powervs [hunk 4]
```diff
@@ -1422,6 +1477,12 @@ function main {
       warn "Enabling forceful destruction option for terraform destroy command"
       FORCE_DESTROY=1
       ;;
+    "-flavor")
+      shift
+      flavortype="$1"
+      vars+=" -var-file ../$varfile -var-file ../automation/compute-vars/$flavortype.tfvars"
+      process_flavor "$flavortype"
```
Comment (yussufsh) [inline openshift-install-powervs:1484]: This should be called before updating the vars.

File: openshift-install-powervs [hunk 5]
```diff
@@ -656,6 +658,22 @@ function precheck_tools {
   fi
 }
 
+#-------------------------------------------------------------------------
+# Check Flavor and run setup
+#-------------------------------------------------------------------------
+function flavor_set {
+
+#  Flavor Option
+  if [ -f "logs/flavor.log" ]; then
```
Comment (yussufsh) [inline openshift-install-powervs:667]: Why new log file?

**Review by yussufsh — CHANGES_REQUESTED** [review]

File: openshift-install-powervs (hunk 1 above)
Comment (aishwaryabk) [reply openshift-install-powervs]: Updated this change.

File: openshift-install-powervs (hunk 4 above)
Comment (aishwaryabk) [reply openshift-install-powervs:1483]: Updated this change

File: openshift-install-powervs (hunk 3 above)
Comment (aishwaryabk) [reply openshift-install-powervs:1382]: A file is used to retain the flavor specified. So that it can retrieve the flavor values in a different run. 
Example: 
$ ./openshift-install-powervs -flavor <option>
$ ./openshift-install-powervs variables

File: openshift-install-powervs (hunk 5 above)
Comment (aishwaryabk) [reply openshift-install-powervs:672]: It checks if the flavor is specified; if not it asks for the  flavor related interactive questions
